### DISCUSSION TITLE
How to use the hybrid extension?

### DISCUSSION THREAD

**Discussion by roastduck** [mkdocstrings/griffe#151] [Q&A] [answered]
Hi,

I highly rely on high-order functions in my package, where some functions are returned by other functions, instead of defined straightforwardly. I see the `hybrid` package is able to dynamically inspect these functions. Is there an example on how to use this extension? I see `hybrid` requires calling another extension to inspect. Can I call the legacy `pytkdocs` to inspect?

Thanks.

**Comment by pawamoy** [comment]
I've started writing some docs but they aren't ready yet. Basically you pass one or more inspector extensions (pytkdocs isn't one) to the hybrid extension, and optionally pass a list of object paths to work on. The inspector extensions then run on the selected objects. I'll try to finish the docs soon :slightly_smiling_face:

**Comment by pawamoy** [comment, marked as answer]
So in the end, I've deprecated the hybrid extension (and more generally the visitor/inspector extension) in favor of a generic `Extension` which is more flexible and robust. See the docs: https://mkdocstrings.github.io/griffe/extensions/, and particularly the full example which can be a good example for your use-case: https://mkdocstrings.github.io/griffe/extensions/#full-example.
